bin/src/utils.dart

```diff
@@ -0,0 +1,239 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/file_system/file_system.dart' as fs;
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:analyzer/source/package_map_resolver.dart';
+import 'package:analyzer/source/pub_package_map_provider.dart';
+import 'package:analyzer/source/sdk_ext.dart';
+import 'package:analyzer/src/dart/sdk/sdk.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/java_io.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer/src/generated/source_io.dart';
+import 'package:path/path.dart' as path;
+
+final RegExp newLinePartOfRegexp = new RegExp('\npart of ');
+final RegExp partOfRegexp = new RegExp('part of ');
+
+Iterable<ClassElement> moduleClassesWithoutNames(
+    AnalysisContext context, Directory sdkDir) {
+  final entryPoints = getPackageEntryPoints();
+  final sources = parseSources(context, entryPoints);
+  verifyAnalysis(context, sources);
+
+  return parseLibraries(context, sources)
+      .expand(getSubclassesOfLifecycleModule)
+      .where(
+          (element) => !element.isAbstract && element.getField('name') == null);
+}
+
+Map<Source, List<ClassElement>> groupClassesBySource(
+    List<ClassElement> classes) {
+  final Map<Source, List<ClassElement>> results = {};
+
+  classes.forEach((c) {
+    if (results.containsKey(c.source)) {
+      results[c.source].add(c);
+    } else {
+      results[c.source] = [c];
+    }
+  });
+
+  return results;
+}
+
+void writeGettersForSource(Source source, List<ClassElement> classes) {
+  // Sort descending to modify the bottom classes before the top
+  classes.sort((a, b) => b.computeNode().end - a.computeNode().end);
+
+  classes.forEach(writeGetterForClass);
+}
+
+void writeGetterForClass(ClassElement e) {
+  final ePath = e.source.uri.path;
+
+  String filePath = ePath.substring(ePath.indexOf('/') + 1);
+
+  File f = new File('lib/$filePath');
+  if (!f.existsSync()) {
+    stdout.writeln('Does not exist: ${f.path}');
+    exit(1);
+  }
+
+  final StringBuffer outputLines = new StringBuffer();
+
+  final source = f.readAsStringSync();
+  // The offset of the end of the class's opening '{' bracket token
+  // ignore: avoid_as
+  final insertionOffset = (e.computeNode() as ClassDeclaration).leftBracket.end;
+
+  outputLines
+    ..writeln(source.substring(0, insertionOffset))
+    ..writeln('  @override')
+    ..writeln('  String get name => \'${e.name}\';')
```

Comment:
Edge-case I just thought of: if a class name contains dollar signs, we'd want to escape them here.

We probably won't ever hit this, so we probably won't need to address it, but just in case you were feeling ambitious...